test/unit/buffer_spec.lua

```diff
@@ -352,6 +364,8 @@ describe('buffer functions', function()
     statusline_test('should truncate at the first `<`', 10,
       'abc%<def%<ghijklm',   'abc<hijklm')
 
+    statusline_test('should ignore trailing %', 3, 'abc%', 'abc')
```

Comment:
`check_stl_option` will report an error if it get this input, so not sure if we really care about this case.